frida_mode/src/instrument/instrument_debug.c

```diff
@@ -1,26 +1,31 @@
+#include <fcntl.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <unistd.h>
 
 #include "frida-gum.h"
 
-#include "util.h"
+#include "debug.h"
 
-#ifdef FRIDA_DEBUG
+#include "util.h"
 
+static int      debugging_fd = -1;
 static gpointer instrument_gen_start = NULL;
 
 static void instrument_debug(char *format, ...) {
 
   va_list ap;
   char    buffer[4096] = {0};
+  int     len = 0;
 
   va_start(ap, format);
 
-  vsnprintf(buffer, sizeof(buffer) - 1, format, ap);
+  len = vsnprintf(buffer, sizeof(buffer) - 1, format, ap);
```

Comment:
I think the return value of vsnprintf is undefined behaviour (because of different implemenation on different platforms. so strlen() afterwards it would need to be.

Reply:
Fixed. There's another one in `stats.c` too.